commands/utils_test.go

```diff
@@ -1,39 +1,290 @@
+			if test.expectedError != "" {
+				require.Error(t, err)
+				assert.EqualError(t, err, test.expectedError)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, test.expectedDetails, result)
+			}
+			assert.Equal(t, "server-id", test.flagProvider.receivedFlagName)
+		})
+	}
+}
+
+func Test_getRtTargetDetails(t *testing.T) {
+	tests := []struct {
+		name            string
+		flagProvider    *flagProviderStub
+		configHelper    serviceHelper
+		sbConf          *supportBundleCommandConfiguration
+		expectedDetails *config.ArtifactoryDetails
+		expectedError   string
+	}{
+		{
+			name:         "default",
+			flagProvider: &flagProviderStub{value: "JFrogSupportLogs"},
+			configHelper: &serviceHelperStub{
+				details: &config.ArtifactoryDetails{
+					Url: "supportlogsurl",
+				},
+			},
+			sbConf: &supportBundleCommandConfiguration{
+				jfrogSupportLogsURL: "supportlogsurl",
+			},
+			expectedDetails: &config.ArtifactoryDetails{
+				Url: "supportlogsurl",
+			},
+		},
+		{
+			name:         "specific target service",
+			flagProvider: &flagProviderStub{value: "my-artifactory"},
+			configHelper: &serviceHelperStub{
+				details: &config.ArtifactoryDetails{
+					Url: "my-artifactory-url",
+				},
+			},
+			sbConf: &supportBundleCommandConfiguration{},
+			expectedDetails: &config.ArtifactoryDetails{
+				Url: "my-artifactory-url/",
+			},
+		},
+		{
+			name:         "get config returns error",
+			flagProvider: &flagProviderStub{value: "my-favorite-artifactory"},
+			configHelper: &serviceHelperStub{
+				configErr: errors.New("oops"),
+			},
+			expectedError: "oops",
+		},
+	}
+
+	for i := range tests {
+		test := tests[i]
+		t.Run(test.name, func(t *testing.T) {
+			result, err := getTargetDetails(test.flagProvider, test.configHelper, test.sbConf)
+			if test.expectedError != "" {
+				require.Error(t, err)
+				assert.EqualError(t, err, test.expectedError)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, test.expectedDetails, result)
+			}
+			assert.Equal(t, "target-server-id", test.flagProvider.receivedFlagName)
+		})
+	}
+}
+
+func Test_parseJSON(t *testing.T) {
+	tests := []struct {
+		name          string
+		payload       string
+		expectedError string
+	}{
+		{
+			name:    "valid json",
+			payload: `{"key":"value"}`,
+		},
+		{
+			name:          "invalid json",
+			payload:       `{"key":"value"`,
+			expectedError: "unexpected end of JSON input",
+		},
+		{
+			name:          "not json",
+			payload:       `"key"`,
+			expectedError: "json: cannot unmarshal string into Go value of type commands.JSONObject",
+		},
+	}
+
+	for i := range tests {
+		test := tests[i]
+		t.Run(test.name, func(t *testing.T) {
+			result, err := parseJSON([]byte(test.payload))
+			if test.expectedError != "" {
+				require.Error(t, err)
+				assert.EqualError(t, err, test.expectedError)
+			} else {
+				require.NoError(t, err)
+				assert.NotEmpty(t, result)
+			}
+		})
+	}
+}
+
+func Test_getJsonString(t *testing.T) {
```

Comment:
Surprised that the linter did not bother you with `Test_getJsonString` instead of `Test_getJSONString`

Reply:
I feel like I got away with something... I almost feel guilty.